src/main/java/org/hydev/mcpm/client/database/inputs/ListPackagesInput.java

```diff
@@ -0,0 +1,15 @@
+package org.hydev.mcpm.client.database.inputs;
+
+/**
+ * Input for the ListAllPackagesBoundary.
+ *
+ * @param itemsPerPage The amount of items for each page.
+ *                     If negative, pageNumber is ignored and all plugins are returned.
+ * @param pageNumber The page number (starting at 0).
+ * @param noCache If true, the ListPackagesBoundary will re-download the database before performing the request.
+ */
+public record ListPackagesInput(
```

Comment:
Maybe add a default no-args constructor, because not all arguments are going to be used.

Reply:
This is an interesting thing to point out. What exactly is the purpose of the list all packages command? I don't think it makes sense for it to be strictly "listing all packages" because the parent has to have access to a DatabaseFetcher object, and listing all packages with that object is as simple as `fetcher.database().plugins()`.